web-console/src/components/table-column-selection.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Button, Checkbox, Popover, Position } from "@blueprintjs/core";
+import * as React from 'react';
+
+import { FormGroup, IconNames } from "./filler";
+
+import "./table-column-selection.scss";
+
+interface TableColumnSelectionProps extends React.Props<any> {
+  tableName: string;
+  onChange: () => void;
+}
+
+interface TableColumnSelectionState {
+  tableColumnSelection: any;
```

Comment:
You can specify a better type here like: `Record<string, boolean>`

Reply:
Actually this component should not even have state... the parent should maintain the state. It is not idiomatic React for the parent to separately pull parse the localStorage.

Here is what you should do: pretend the browser localStorage is crippled (see `localStorageGet` function), this component should still work - it would just not save the state between refreshes